modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java

```diff
@@ -466,6 +495,25 @@ public void stopBlockMessages(String groupId) {
         client.stopBlock();
     }
 
+    /**
+     * Initialize (if needed) and receive per raft group lock,
+     * to prevent concurrent start of the same raft group.
+     *
+     * @param grpId Group id.
+     * @return lock
+     */
+    private Lock groupLock(String grpId) {
+        var newGroupLock = new ReentrantLock();
```

Comment:
It seems to me that this is inefficient to create new ReentrantLock every time we call this method. Let's create it only if `startGroupInProgressLocks` doesn't contain lock for grpId

Reply:
Reworked the whole locking mechanism - now it is simplier and without these issues